casadm/cas_lib.c

```diff
@@ -1849,6 +1825,37 @@ int remove_core(unsigned int cache_id, unsigned int core_id,
 	return SUCCESS;
 }
 
+int check_cache_state_incomplete(int cache_id, int fd) {
+	int i, status = SUCCESS;
+	struct cache_device *cache;		
+	
+	for (i = 0; i < CORE_ADD_MAX_TIMEOUT; i++) {
+		cache = get_cache_device_by_id_fd(cache_id, fd, false);
+		status = FAILURE;
+		
+		if (cache == NULL) {
+			break;	
+		}
+
+		if (cache->core_count == cache->expected_core_count) {
+			if (cache->state & (1 << ocf_cache_state_incomplete)) {
+				cas_printf(LOG_WARNING, "WARNING: Cache is in incomplete state - at least one core is inactive\n");
+			}
+			status = SUCCESS;
+			free(cache);
+			cache = NULL;
+			break;
+		}
+	
+		free(cache);
+		cache = NULL;
+
+		sleep(1);
+	}
+
+	return status;
```

Comment:
You never check the return value from this function so status is unnecessary and function return type may be converted into void